scripts/welding_schema.ipynb

```diff
@@ -0,0 +1,485 @@
+    ")\n",
+    "\n",
+    "tcp_start_point = Q_([5.0, 0.0, 2.0], \"mm\")\n",
+    "tcp_end_point = Q_([-5.0, 0.0, 2.0], \"mm\") + np.append(seam_length, Q_([0, 0], \"mm\"))\n",
+    "\n",
+    "v_weld = Q_(10, \"mm/s\")\n",
+    "s_weld = (tcp_end_point - tcp_start_point)[0]  # length of the weld\n",
+    "t_weld = s_weld / v_weld\n",
+    "\n",
+    "t_start = pd.Timedelta(\"0s\")\n",
+    "t_end = pd.Timedelta(str(t_weld.to_base_units()))\n",
+    "\n",
+    "rot = WXRotation.from_euler(seq=\"x\", angles=180, degrees=True)\n",
+    "\n",
+    "coords = [tcp_start_point.magnitude, tcp_end_point.magnitude]\n",
+    "\n",
+    "tcp_wire = lcs(coordinates=coords, orientation=rot, time=[t_start, t_end])\n",
+    "\n",
+    "# add the workpiece coordinate system\n",
+    "csm.add_cs(\n",
+    "    coordinate_system_name=\"tcp_wire\",\n",
+    "    reference_system_name=\"workpiece\",\n",
+    "    lcs=tcp_wire,\n",
+    ")\n",
+    "\n",
+    "tcp_contact = lcs(coordinates=[0, 0, -10])\n",
+    "\n",
+    "# add the workpiece coordinate system\n",
+    "csm.add_cs(\n",
+    "    coordinate_system_name=\"tcp_contact\",\n",
+    "    reference_system_name=\"tcp_wire\",\n",
+    "    lcs=tcp_contact,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Measurements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# time\n",
+    "time = pd.timedelta_range(start=\"0s\", end=\"10s\", freq=\"1ms\")\n",
+    "\n",
+    "# current data\n",
+    "I_ts = ut.sine(f=Q_(10, \"1/s\"), amp=Q_(20, \"A\"), bias=Q_(300, \"A\"))\n",
+    "I = I_ts.interp_time(time)\n",
+    "I[\"time\"] = I[\"time\"]\n",
+    "\n",
+    "current_data = msm.Data(name=\"Welding current\", data=I)\n",
+    "\n",
+    "# voltage data\n",
+    "U_ts = ut.sine(f=Q_(10, \"1/s\"), amp=Q_(3, \"V\"), bias=Q_(40, \"V\"), phase=Q_(0.1, \"rad\"))\n",
+    "U = U_ts.interp_time(time)\n",
+    "U[\"time\"] = U[\"time\"]\n",
+    "\n",
+    "voltage_data = msm.Data(name=\"Welding voltage\", data=U)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from asdf.tags.core import Software\n",
+    "\n",
+    "HKS_sensor = msm.GenericEquipment(name=\"HKS P1000-S3\")\n",
+    "BH_ELM = msm.GenericEquipment(name=\"Beckhoff ELM3002-0000\")\n",
+    "twincat_scope = Software(name=\"Beckhoff TwinCAT ScopeView\", version=\"3.4.3143\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "src_current = msm.Source(\n",
+    "    name=\"Current Sensor\",\n",
+    "    output_signal=msm.Signal(signal_type=\"analog\", unit=\"V\", data=None),\n",
+    "    error=msm.Error(Q_(0.1, \"percent\")),\n",
+    ")\n",
+    "\n",
+    "HKS_sensor.sources = []\n",
+    "HKS_sensor.sources.append(src_current)\n",
```

Comment:
Why not in a single statement? Don't want to be picky since this is not meant as a tutorial. Just curious :p - Same goes for line 236 and 237